modules/lightning_features/lightning_core/lightning_core.module

```diff
@@ -81,6 +110,50 @@ function lightning_core_disable_buttons(array $element) {
 }
 
 /**
+ * Implements hook_form_alter().
+ */
+function lightning_core_form_alter(array &$form, FormStateInterface $form_state) {
+  $form_object = $form_state->getFormObject();
+
+  if ($form_object instanceof EntityFormInterface) {
+    $entity = $form_object->getEntity();
+
+    if ($entity instanceof EntityDisplayModeInterface || $entity instanceof RoleInterface) {
+      $form['description'] = [
+        '#type' => 'textarea',
+        '#title' => t('Description'),
+        '#description' => t('Additional relevant information about this @entity_type, such as where it is used and what it is for.', [
+          '@entity_type' => $entity->getEntityType()->getSingularLabel(),
+        ]),
+        '#default_value' => $entity->getThirdPartySetting('lightning_core', 'description'),
+      ];
+      $form['actions']['submit']['#submit'][] = 'lightning_core_set_third_party_description';
+    }
+  }
+}
+
+/**
+ * Form submit handler: updates the description of a configuration entity.
+ *
+ * @param array $form
+ *   The complete form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current form state.
+ */
+function lightning_core_set_third_party_description(array &$form, FormStateInterface $form_state) {
+  $form_state
+    ->getFormObject()
+    ->getEntity()
+    ->setThirdPartySetting('lightning_core', 'description', $form_state->getValue('description'))
+    ->save();
+
+  // Help text is render cached, so clear the entire render cache right now so
+  // it will pick up the updated description. @see lightning_core_help()
+  // TODO: Try to use cache tags here so as to be less of a blunt instrument.
```

Comment:
Let's go ahead and do that. Give the help block it's own cache tag and then just clear that. I could see some editor on rod making a tweak here and bringing the site to its knees.